js/src/sdk/utils/pusher.ts

```diff
@@ -78,43 +104,41 @@ export class PusherUtils {
    * Binds an event to a channel with support for chunked messages.
    * @param {PusherClient} channel - The Pusher channel to bind the event to.
    * @param {string} event - The event to bind to the channel.
-   * @param {(data: any) => void} callback - The callback function to execute when the event is triggered.
+   * @param {(data: unknown) => void} callback - The callback function to execute when the event is triggered.
    */
   private static bindWithChunking(
     channel: PusherClient,
     event: string,
-    callback: (data: any) => void
+    callback: (data: Record<string, unknown>) => void
   ): void {
     channel.bind(event, callback); // Allow normal unchunked events.
 
     // Now the chunked variation. Allows arbitrarily long messages.
     const events: {
       [key: string]: { chunks: string[]; receivedFinal: boolean };
     } = {};
-    channel.bind(
-      "chunked-" + event,
-      (data: { id: string; index: number; chunk: string; final: boolean }) => {
-        if (!events.hasOwnProperty(data.id)) {
-          events[data.id] = { chunks: [], receivedFinal: false };
-        }
-        const ev = events[data.id];
-        ev.chunks[data.index] = data.chunk;
-        if (data.final) ev.receivedFinal = true;
-        if (
-          ev.receivedFinal &&
-          ev.chunks.length === Object.keys(ev.chunks).length
-        ) {
-          callback(JSON.parse(ev.chunks.join("")));
-          delete events[data.id];
-        }
+    channel.bind("chunked-" + event, (data) => {
+      const typedData = data as TChunkedTriggerData;
+      if (!events.hasOwnProperty(typedData.id)) {
```

Comment:
Consider adding error handling for the case where `JSON.parse()` fails when joining chunks. A malformed chunk could cause the application to crash. Example:
```typescript
try {
  callback(JSON.parse(ev.chunks.join("")));
} catch (error) {
  logger.error(`Failed to parse chunked message: ${error}`);
  // Handle the error appropriately
}
```